llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -23429,17 +23429,21 @@ SDValue DAGCombiner::visitBUILD_VECTOR(SDNode *N) {
   // TODO: Maybe this is useful for non-splat too?
   if (!LegalOperations) {
     if (SDValue Splat = cast<BuildVectorSDNode>(N)->getSplatValue()) {
```

Comment:
The old code seems problematic, if we use `cast`, we don't need to use `if`.

Reply:
Probably change it to dyn_cast?

Reply:
Consider not assertion happened before, removing the `if` is also fine.